singer-io/tap-github
# TDL-12323 Implemented currently syncing for repos and streams

# Description of change
- Implemented the currently syncing repos and streams
- Added unittests and tap-tester tests for the same

# Manual QA steps
 - Verify that the sync works properly when a new stream is selected in the catalog after the sync is interrupted.
 - Verify that the sync works properly when a selected stream is deselected in the catalog after the sync is interrupted.
 - Verify that the sync works properly when a currently syncing stream is deselected in the catalog after the sync is interrupted.
 - Verify that all the records respect the current syncing stream's bookmark.
 - Verify the order is maintained for the interrupted sync stream/repo i.e. the interrupted stream/repo will be synced first.
 
# Risks
 - 
 
# Rollback steps
 - revert this branch
